services/src/datasets/storage.rs

```diff
@@ -18,7 +19,21 @@ pub struct DataSet {
     pub name: String,
     pub description: String,
     pub data_type: LayerInfo, // TODO: custom type instead of reusing existing one?
-    pub source_operator: String, // TODO: enum?
+    pub source_operator: String,
+    pub spatial_reference: SpatialReferenceOption,
+}
+
+impl DataSet {
+    pub fn listing(&self) -> DataSetListing {
+        DataSetListing {
+            id: self.id.clone(),
+            name: self.name.clone(),
+            description: self.description.clone(),
```

Comment:
is it actually necessary to clone all this stuff or could we just give out &str? We serialize it anyway.

Reply:
it's not necessary but I use this function only in the in memory DataSetDB. In real deployments using Postgres backend I would create the DataSetListing directly out of data from the db and then the Listing actually has to own the data